doc/voq/voq_hld.md

```diff
@@ -0,0 +1,879 @@
+1.  Packets that are trapped to the host asic namespace because of an exception (unresolved ARP for example)
+2.  Packets that are sent to the host asic namespace because the destination address on the packet is an interface address for SONiC on the asic and
+3.  All packets sent from the host asic namespace and out via the network ports of the asic.
+
+In a Distributed VOQ System, the requirement for network communication for the asic namespace remains the same. But there are some important differences.
+1.  Packets that are trapped to the host asic namespace because of an exception - These are only received on network ports of the local asic
+2.  Packets that are sent to the host asic namespace because the destination address on the packet is an interface address for SONiC on the asic. These could be
+    - Received on a network port of the local asic OR
+    - Received on a network port of another asic in the system and sent to the local asic over the fabric interfaces OR
+    - Received on the CPU port of another asic in the system and sent to the local asic over the fabric interfaces
+3.  Packet sent from host asic namespace could be either
+    - Sent out via a network port of the local asic OR
+    - Sent over the fabric interfaces to another asic and out via a network port of that asic OR
+    - Sent over the fabric interfaces to another asic and out via the CPU port of that asic (to asic namespace of the other asic)
+
+The figure below shows the cpu-to-cpu packet flows that needs to be supported
+ ![](../../images/voq_hld/cpu-to-cpu-packet-flow.png)
+
+The figure below shows the cpu-to-network-port packet flows that need to be supported
+ ![](../../images/voq_hld/network-to-cpu-packet-flows.png)
+
+## 1.6 CLI requirements
+User should be able to display/retrieve the following VOQ information
+- system-port id for the system ports
+- Switch-id
+- max-cores
+- Neighbor Encap index
+
+## 1.7 Scalability requirements
+
+###### Table 2: VOQ scalability
+| VOQ component            | Expected value              |
+|--------------------------|-----------------------------|
+| VOQ Switches             | 128                         |
+| System Ports             | 4K                          |
+| VOQ Neighbors            | 16k                         |
+
+The initial scaling should at least support the requirements of a "large" modular chassis deploying ports based routing interfaces. Virtual interfaces (VLAN based routing interfaces) and directly attached hosts are still supported, but achieving higher scaling for those deployments is not a goal of the initial implementation.
+
+## 1.8 Warm Restart requirements
+At this time warm restart capability is not factored into the design. This shall be revisited as a Phase #2 enhancement
+
+# 2 Modules Design
+## 2.1 Configuration Flow Options for System Ports
```

Comment:
The approach of using FSI CONFIG_DB to capture the global configuration including system ports (and LAGs in the future) will be documented in the Arch doc pull/622. The details including the schema can be presented here.